src/store/permissions.ts

```diff
@@ -0,0 +1,158 @@
+import type { ApplicationCommandPermissionData } from "discord.js";
+import type { State } from "../state";
+
+interface PermissionTree {
+	[key: string]: PermissionObject;
+}
+
+interface PermissionObject {
+	field: string;
+	perms: ApplicationCommandPermissionData[];
+}
+
+let state: State | undefined;
+
+export function setPermState(newState: State): void {
+	state = newState;
+}
+
+function getState() {
+	return state;
+}
+
+export function getPermissions(commandName: string): PermissionObject {
+	return permissions[commandName];
+}
+
+const permissions: PermissionTree = {
+	cmdhistory: {
+		field: "staffRoleId",
+		perms: [
```

Comment:
`getState()` in the following set of options will still return undefined, effectively making all the IDs "0".